### ISSUE TITLE
1.18.2 Invalid moving sound AIR_LEAK

### ISSUE THREAD

**Issue by Raidobw2** [TeamPneumatic/pnc-repressurized#1046]
### Describe the bug

I created a floating platform of portals in the AE2 Spatial IO dimension to serve as our hub. At the same time my friend was progressing in PneumaticCraft and placed a few Safety Tube Modules which frequently play the AIR_LEAK sound. Using these portals often leads to a client-side crash:
```
java.lang.IllegalArgumentException: Invalid moving sound AIR_LEAK for focus object BlockPos{x=154, y=79, z=204}
```
The coordinates in each crash report point to a PneumaticCraft Safety Tube Module, which plays the AIR_LEAK sound every few seconds.

### Versions:
Minecraft `1.18.2`
Forge `40.1.52`
PneumaticCraft `pneumaticcraft-repressurized-1.18.2-3.3.0-99.jar`
Mekanism `Mekanism-1.18.2-10.2.4.464.jar`
Modpack `FTB Plexiglass Mountain 1.1.1`

### How to reproduce the bug

1. Create a Mekanism Teleporter (unclear if it needs to be in/out of Spatial IO or not).
2. Attach a Safety Tube Module to a pressure tube. We are using a Flux Compressor w/ 4 speed upgrades to pressurize the tube.
3. Use the teleporter until the client crashes.

### Expected behavior

No client crashes while using our teleporters

### Additional details

Relayed from https://github.com/FTBTeam/FTB-Modpack-Issues/issues/1071

![image](https://user-images.githubusercontent.com/17010632/177882850-a90bb297-ed17-4b2a-99f5-b8f247dbb1fe.png)
![image](https://user-images.githubusercontent.com/17010632/177882871-6ea8b921-2ff6-40ff-879b-ca66c878b21d.png)

Reading the code in `MovingSounds.java` for PNC, my guess is that this does have something to do with the portals crossing into different dimensions:

https://github.com/TeamPneumatic/pnc-repressurized/blob/1.18.2/src/main/java/me/desht/pneumaticcraft/client/sound/MovingSounds.java#L84-L89

The `BlockEntity te = world.getBlockEntity(pos);` assignment may be returning `null` because it is checking for the Safety Tube Module in the wrong dimension as the player teleports. It decides there is nothing there to play a sound for, breaks out of the `switch` and throws the exception. It seems like this could happen any time a player enters a new dimension and the sound is trying to play.

Thanks in advance!

### Which Minecraft version are you using?

1.18

### Which version of PneumaticCraft: Repressurized are you using?

1.18.2-3.3.0-99

### Crash log

https://paste.feed-the-beast.com/view/76912281

**Comment by desht** [comment]
Yeah, I see the problem there. An exception should only be thrown if the packet data is downright invalid, but in this case the packet data is valid but the chunk isn't fully loaded clientside yet (which is why the block entity appears to be missing). In this case null needs to be returned indicating no sound to be played yet.

**Comment by desht** [comment]
The https://github.com/TeamPneumatic/pnc-repressurized/actions/runs/2634619301 build should fix this, feel free to test.  The fix will be in the next release (3.3.2).

**Comment by zarillion** [comment]
Thanks for the quick turn-around!

**Comment by desht** [comment]
Fixed in 3.4.0 release

**Comment by Raidobw2** [comment]
Awesome, thank you!
